optimum/onnxruntime/modeling_seq2seq.py

```diff
@@ -383,6 +388,33 @@ def forward(
 
         return BaseModelOutput(last_hidden_state=last_hidden_state)
 
+    def compute_encoder_known_output_shapes(self, pixel_values: torch.FloatTensor) -> Dict[str, List[int]]:
+        if self.normalized_config.model_type == "vit":
+            # for vit models
+            encoder_sequence_length = (
+                self.normalized_config.image_size // self.normalized_config.config.patch_size
+            ) ** 2 + 1  # plus cls token
+        elif self.normalized_config.config.model_type == "donut-swin":
+            # for donut-swin models
+            encoder_sequence_length = (
+                self.normalized_config.config.image_size[0]
+                * self.normalized_config.config.image_size[1]
+                // self.normalized_config.config.hidden_size
+            )
+        else:
+            raise ValueError(
+                f"Unsupported encoder model type {self.normalized_config.config.model_type} for VisionEncoderDecoder."
```

Comment:
This is an other case where we may benefit from subclassing, I'll try to do a PR rather soon than later.